File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    const n = s.length;
    let start = 0, maxLen = 1;
    for (let i = 0; i < n; i++) {
        let left = i, right = i;
        while (left >= 0 && right < n && s[left] === s[right]) {
            if (right - left + 1 > maxLen) {
                start = left;
                maxLen = right - left + 1;
            }
            left--;
            right++;
        }
        left = i;
        right = i + 1;
        while (left >= 0 && right < n && s[left] === s[right]) {
            if (right - left + 1 > maxLen) {
                start = left;
                maxLen = right - left + 1;
            }
            left--;
            right++;
        }
    } 
    return s.substring(start, start + maxLen);
};

File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.erl
-spec longest_palindrome(S :: unicode:unicode_binary()) -> unicode:unicode_binary().
longest_palindrome(S) ->
    Chars = unicode:characters_to_list(S),
    case Chars of
        [] -> <<>>;
        [C] -> unicode:characters_to_binary([C]);
        _ ->
            Transformed = ['^' | lists:append(['#'], lists:flatmap(fun(C) -> [C, '#'] end, Chars)) ++ ['$']],
            {Center, MaxLen} = manacher(Transformed),
            Start = (Center - MaxLen + 1) div 2,
            Length = MaxLen,
            unicode:characters_to_binary(lists:sublist(Chars, Start, Length))
    end.
manacher(Chars) ->
    Len = length(Chars),
    P = array:new(Len, {default, 0}), % Array for palindrome lengths
    manacher_helper(Chars, P, 1, 0, 0, {0, 0}).
manacher_helper(Chars, P, I, C, R, {MaxC, MaxL}) when I >= length(Chars) ->
    {MaxC, MaxL};
manacher_helper(Chars, P, I, C, R, {MaxC, MaxL}) ->
    P_I = if
        I =< R ->
            Mirror = 2 * C - I,
            erlang:min(R - I, array:get(Mirror, P));
        true -> 0
    end,
    {NewP_I, NewP} = expand_palindrome(Chars, P, I, P_I),
    {NewC, NewR} = if
        I + NewP_I > R -> {I, I + NewP_I};
        true -> {C, R}
    end,
    NewMax = if
        NewP_I > MaxL -> {I, NewP_I};
        true -> {MaxC, MaxL}
    end,
    manacher_helper(Chars, NewP, I + 1, NewC, NewR, NewMax).
expand_palindrome(Chars, P, I, P_I) ->
    Left = I - P_I - 1,
    Right = I + P_I + 1,
    case Left >= 1 andalso Right =< length(Chars) andalso lists:nth(Left, Chars) =:= lists:nth(Right, Chars) of
        true -> expand_palindrome(Chars, P, I, P_I + 1);
        false -> {P_I, array:set(I, P_I, P)}
    end.